docs(notification): add kb for keeping notifications visible on hover (#1735)
telerik/blazor-docs: knowledge-base/notification-keep-on-hover.md

---
title: Prevent Notification Hiding on Hover
description: How to stop a notification from closing automatically when the user hovers it?
type: how-to
page_title: How to Prevent Notification Hiding on Mouse Over
slug: notification-kb-keep-on-hover
position: 
tags: telerik, blazor, notification
ticketid: 1630136
res_type: kb
---

## Environment

<table>
    <tbody>
        <tr>
            <td>Product</td>
            <td>Notification for Blazor</td>
        </tr>
    </tbody>
</table>

## Description

This KB article answers the following questions:

* Is it possible to stop a notification from closing on mouseover?
* How to keep a notification from hiding automatically when the user moves their mouse over it?
* How to persist a visible notification on the screen when the user hovers it?
* How to prevent hiding of a notification when the mouse cursor is over it?

## Solution

1. [`Show()` notifications]({%slug notification-open-close-hide%}) that don't close automatically. To achive this, set `CloseAfter` to `0` in the `NotificationModel`.

> 1. Define a collection that will store the currently open `NotificationModel` instances and the `DateTime` values when they should hide.

Reviewer: ```suggestion
1. Define a collection that will store the currently open `NotificationModel` instances and the `DateTime` values that determine when they will be hidden.
```

> 1. Implement a `System.Timers.Timer` that will [`Hide()` notifications]({%slug notification-open-close-hide%}), according to the times from the previous step.

Reviewer: ```suggestion
1. Implement a `System.Timers.Timer` that will [`Hide()` notifications]({%slug notification-open-close-hide%}) according to the times from the previous step.
```

1. Define a [Notification `Template`]({%slug notification-templates%}).

> 1. Wrap the template content in a `<div>` with [`@onmouseenter` and `@onmouseleave` directives](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling). The event handlers should enable or disable the hiding of that specific notification instance by the `Timer`.

Reviewer: ```suggestion
1. Wrap the template content in a `<div>` with [`@onmouseenter` and `@onmouseleave` directives](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling). The event handlers must enable or disable the hiding of that specific notification instance by the `Timer`.
```
